src/verify_cert.rs

```diff
@@ -20,73 +20,104 @@ use pki_types::{CertificateDer, SignatureVerificationAlgorithm, TrustAnchor};
 use crate::cert::Cert;
 use crate::crl::RevocationOptions;
 use crate::der::{self, FromDer};
+use crate::end_entity::EndEntityCert;
+use crate::time::Time;
 use crate::{signed_data, subject_name, time, Error};
 
-pub(crate) struct ChainOptions<'a> {
-    pub(crate) eku: KeyUsage,
-    pub(crate) supported_sig_algs: &'a [&'a dyn SignatureVerificationAlgorithm],
-    pub(crate) trust_anchors: &'a [TrustAnchor<'a>],
-    pub(crate) intermediate_certs: &'a [CertificateDer<'a>],
-    pub(crate) revocation: Option<RevocationOptions<'a>>,
+/// Input context for verifying a certificate chain for a particular purpose.
+pub struct ChainOptions<'a> {
+    /// Intended usage of the certificate, indicating what purpose verification is for.
+    pub eku: KeyUsage,
+    /// Signature algorithms that are trusted for use in certificate signatures.
+    ///
+    /// The end-entity certificate's public key is not validated against this list.
+    pub supported_sig_algs: &'a [&'a dyn SignatureVerificationAlgorithm],
+    /// List of root CAs that will anchor the trust in the built path.
+    pub trust_anchors: &'a [TrustAnchor<'a>],
+    /// Intermediate certificates that a peer sent for the purpose of path building.
+    pub intermediate_certs: &'a [CertificateDer<'a>],
+    /// Certificate revocation list options to optionally check for revocation.
+    pub revocation: Option<RevocationOptions<'a>>,
 }
 
 impl<'a> ChainOptions<'a> {
-    pub(crate) fn build_chain(&self, cert: &Cert<'_>, time: time::Time) -> Result<(), Error> {
-        let path = PathNode { cert, issued: None };
-        self.build_chain_inner(&path, time, 0, &mut Budget::default())
-            .map_err(|e| match e {
-                ControlFlow::Break(err) => err,
-                ControlFlow::Continue(err) => err,
-            })
+    /// Verifies that the end-entity certificate is valid for use against the
+    /// specified Extended Key Usage (EKU).
+    ///
+    /// `time` is the time for which the validation is effective (usually the
+    /// current time).
+    pub fn verify_trusted(
```

Comment:
I think we should update the rustdoc for this fn to discuss the `verify_path` arg.